### PR TITLE
change setup.py to poetry

### PR DESCRIPTION
**Pull request by Isafagioli-Tenchi**
changed files to delete setup.py and use poetry

### CODE REVIEW

**Review by edpin — DISMISSED** [review]
LG by me, but I'm not the expert here. Added folks who might know more.

**Review by asieira — CHANGES_REQUESTED** [review]
The pyproject.toml file does not migrate all settings that used to be part of setup.py. For example I didn't see https://python-poetry.org/docs/pyproject/#classifiers being covered at all. Please review ALL settings in setup.py and migrate them to their Poetry equivalents.

**Comment by asieira** [comment]
Also, did we try uploading this to PyPItest and then install from PyPItest in a container or VM to ensure the end-to-end publishing and installation process still works for customers using pip? We should.

**Comment by asieira** [comment]
Also, let's make sure the PyPItest page matches the classifiers, documentation and links as per the main PyPI one we used to have using pip and setup.py.

**Comment by RomuloGatto** [comment]
@asieira All of your need were matched and tested by Isa and I, please take a look.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:25]: This is incorrect, we shouldn't hard code the version. The previous code obtained the version to be used from the Github tag of the release, and we should continue doing so.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:26]: Same here.

**Review by asieira — CHANGES_REQUESTED** [review]
I see you published a new version of the CLI in PyPITest yesterday, but something is wrong since no description is available.
image

See the "proper" PyPI listing for the CLI for what is expected there.
image

The old deployment process in the makefile and GitHub actions using setup.py imported the README and an auto-generated documentation of the CLI commands and parameters. This needs to be fixed.

**Comment by RomuloGatto** [comment]
@asieira fixed all the version typos. The missing readme was already fixed, but I have redeployed it to be sure, please take a look. https://test.pypi.org/project/zanshincli/1.34.1/

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:29]: Nope, the hard coding is still here.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:30]: Same.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:31]: This is the code you want, but it's commented. Seems like you committed the local code in the way you set it up for testing, but I can't approve the PR like this.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:66]: This also seems to be uncommented before we can approve the PR.

**Review by asieira — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml
Comment (RomuloGatto) [reply .github/workflows/release.yml]: > This is the code you want, but it's commented. Seems like you committed the local code in the way you set it up for testing, but I can't approve the PR like this.

I missed the push button haha, now it's uploaded.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:58]: Is there a way for us to set the PyPI and TestPyPI passwords on the environment instead of on the command-line?

Here is the problem - the commands executed will probably show up in the Github Actions execution history, and that might expose the password. 🤔  Safer to set it to an environment variable that will be then used directly by poetry.

File: .github/workflows/release.yml
Comment (asieira) [inline .github/workflows/release.yml:63]: Same here.

**Review by asieira — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml
Comment (asieira) [reply .github/workflows/release.yml:58]: Turns out Github handles that automatically and masks the password, which is awesome:
image

So no change needed here.

**Review by asieira — APPROVED** [review]
LGTM

**Review by RomuloGatto — APPROVED** [review]

**Pull request merged**
